app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php

```diff
@@ -98,23 +105,23 @@ protected function validateGuestSubscription()
      * Validates the format of the email address
      *
      * @param string $email
-     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws LocalizedException
+     * @throws \Zend_Validate_Exception
```

Comment:
Looks like this type of exception should not be thrown. Am I wrong?

Reply:
I got the message from PHPStorm that not all the necessary `@throws` tags were set, so I added this with the autocomplete in PHPStorm.

Reply:
AFAIK `\Zend_Validate_Exception` could be thrown in case if validator wasn't found. So this method actually can't throw such exceptions.
Please remove it.
Everything else looks good.

Reply:
In **vendor/magento/zendframework1/library/Zend/Validate.php** in the static `is` method, not all exceptions are being caught. So I should leave the `@throws` tag or write a `try/catch`.

Please see: https://github.com/magento/zf1/blob/48ac5e38c591afefb95e0d7127d00fe52d0fbb0f/library/Zend/Validate.php#L195

Reply:
@ihor-sviziev I found out that Magento actually had an email validator class, I required this one through dependency injection. This saves some reflection :)